### PR TITLE
out_gcs: implement parquet compressions

### PR DESCRIPTION
**Pull request by cosmo0920**
## Summary

- Add Parquet output support to the GCS output plugin.
- Support `none`, `snappy`, `gzip`, and `zstd` Parquet page compression.
- Set the default Parquet content type to `application/vnd.apache.parquet`.
- Reject Parquet configuration when Fluent Bit is built without Parquet support.
- Publish the shared columnar compression function under the generic AWS compression API while retaining the old S3 function for compatibility.
- Update S3 and internal tests to use the public API.
- Add GCS runtime coverage for Parquet uploads and unsupported builds.

## Testing

- Ran the unit test suite:

  ```console
  ctest -j 18 . --output-on-failure
  ```

- Manually confirmed GCS Parquet output works with:

  ```yaml
  format: parquet
  compression: gzip # also tested: snappy, zstd, none
  ```

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

## Summary by CodeRabbit

* **New Features**
  * Added Parquet output support for Google Cloud Storage.
  * Added Parquet compression options: none, Snappy, GZIP, and Zstandard.
  * Added JSON Lines output with none or GZIP compression.
  * Applied appropriate content types and upload handling for each format.

* **Bug Fixes**
  * Improved validation and error reporting when Parquet support is unavailable.
  * Preserved compatibility with existing S3 columnar compression behavior.

### CODE REVIEW

File: src/aws/compression/arrow/compress.c [hunk 1]
```diff
@@ -239,9 +239,10 @@ static GArrowResizableBuffer* table_to_parquet_buffer(GArrowTable *table,
 }
 #endif
 
-int out_s3_compress_columnar(int columnar_format, void *json, size_t size,
-                             void **out_buf, size_t *out_size,
-                             int compression_type)
+int flb_aws_compression_compress_columnar(int columnar_format,
+                                          void *json, size_t size,
+                                          void **out_buf, size_t *out_size,
+                                          int compression_type)
```
[the remark this answers is not in the document]
Comment (cosmo0920) [reply src/aws/compression/arrow/compress.c:245]: This review comment should be addressed in another PR.
I sent a PR which is built on top of this PR: https://github.com/fluent/fluent-bit/pull/12304
This PR can address the concern.
